invenio_records_resources/services/files/service.py

```diff
@@ -95,72 +95,85 @@ def update_file_metadata(self, identity, id_, file_key, data, uow=None):
         """Update the metadata of a file."""
         record = self.get_record(id_, identity, "create_files")
 
-        self.run_components(
-            "update_file_metadata", identity, id_, file_key, record, data, uow=uow
-        )
-
-        return self.file_result_item(
-            self,
-            identity,
-            record.files[file_key],
-            record,
-            links_tpl=self.file_links_item_tpl(id_),
-        )
+        if file_key in record.files:
+            self.run_components(
+                "update_file_metadata", identity, id_, file_key, record, data, uow=uow
+            )
+
+            return self.file_result_item(
+                self,
+                identity,
+                record.files[file_key],
+                record,
+                links_tpl=self.file_links_item_tpl(id_),
+            )
+        else:
+            return None
 
     def read_file_metadata(self, identity, id_, file_key):
         """Read the metadata of a file."""
         record = self.get_record(id_, identity, "read_files")
 
-        self.run_components("read_file_metadata", identity, id_, file_key, record)
+        if file_key in record.files:
```

Comment:
minor, could improve readability
```suggestion
        if file_key not in in record.files:
            return None
```